src/fat_beagle.cpp

```diff
@@ -65,17 +66,18 @@ double FatBeagle::LogLikelihoodInternals(const Tree &tree) const {
 }
 
 double FatBeagle::LogLikelihood(const Tree &tree) const {
-  return LogLikelihoodInternals(DetrifurcateIfNeeded(tree));
+  return LogLikelihoodInternals(DetrifurcateIfNeeded(tree).Topology(),
+                                tree.BranchLengths());
```

Comment:
This seems dangerous to me. If we _do_ detrifurcate a tree, then the branch lengths will be out of sync. 

It seems like we want 

```
auto detrifurcated_tree = DetrifurcateIfNeeded(tree);
return LogLikelihoodInternals(detrifurcated_tree.Topology(),
                                detrifurcated_tree.BranchLengths());
```

Reply:
good point!